learning_journal/views/default.py

```diff
@@ -58,10 +62,38 @@ def single_entry(request):
 
 @view_config(
     route_name='edit-entry',
-    renderer='learning_journal:templates/edit-entry.html'
+    renderer='learning_journal:templates/edit-entry.html',
+    permission='modify'
 )
 def edit_entry(request):
     """Render edit-entry.html at '/journal/12345/edit-entry'"""
     return {
         'title': 'Edit entry'
     }
+
+
+@view_config(
+    route_name='login',
+    renderer='../templates/login.html'
+)
+def login(request):
+    error = ''
+    if request.method == 'POST':
+        username = request.params.get('username', '')
+        password = request.params.get('password', '')
+        print(username, password, check_credentials(username, password))
```

Comment:
again, kill off debugging code before pushing to github.  The habit will save you from accidentally pushing code that contains break points or other bad things.